src/App.tsx

```diff
@@ -15,48 +18,63 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getPreparedGoods(
+  goods: string[],
+  method: SortType,
+  isReversedOrder: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (method) {
+    switch (method) {
+      case (SortType.Alphabet): {
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      }
+
+      case (SortType.Length): {
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      }
+
+      default:
+        break;
+    }
+  }
+
+  if (isReversedOrder) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+  const goods = getPreparedGoods(goodsFromServer, sortBy, isReversed);
+
+  const ChangeSortMethods = (m:SortType) => {
+    setSortBy(m);
+  };
+
+  const SetSortByDefault = () => {
+    setSortBy(SortType.Default);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+      <SortBar
+        sortBy={sortBy}
+        ChangeSortMethods={ChangeSortMethods}
+        SetSortByDefault={SetSortByDefault}
```

Comment:
please update to camelCase
`ChangeSortMethods` change to `changeSortMethods`
`SetSortByDefault` change to `setSortByDefault`

change across all your files